### PR TITLE
Use upstream oniguruma

### PR DESCRIPTION
**Pull request by Ingramz**
Oniguruma moved a while ago to GitHub (https://github.com/kkos/oniguruma), let's use that.

This should make updating the dependency from upstream easier and perhaps if something doesn't compile due to oniguruma, we can perhaps get it fixed upstream, if applicable. Only thing that's custom from that now is the config file for non-windows targets.

Tested with VS2015

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "deps/onig"]
+	path = deps/onig
```
Comment (mnquintana) [inline .gitmodules]: This is a bit of a nitpick, but would you mind using 2-space indentation here instead of tabs? :bow:

Comment (Ingramz) [inline .gitmodules]: That file was autogenerated by git and I don't see the two-space convention used anywhere else inside `.gitmodules` in the atom project. If you think it's still worth doing, then sure, no problem.

Comment (mnquintana) [inline .gitmodules]: Ah never mind then :+1:

**Comment by Ingramz** [comment]
Closing this pull request due to lack of interest.

**Pull request closed without merging**